AutoCollection/ClientRequests.ts

```diff
@@ -0,0 +1,78 @@
+///<reference path="..\Declarations\node\node.d.ts" />
+
+import http = require("http");
+import https = require("https");
+import url = require("url");
+
+import ContractsModule = require("../Library/Contracts");
+import Client = require("../Library/Client");
+import Logging = require("../Library/Logging");
+import Util = require("../Library/Util");
+import ClientRequestParser = require("./ClientRequestParser");
+
+class AutoCollectClientRequests {
+
+    public static INSTANCE: AutoCollectClientRequests;
+
+    private _client:Client;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+
+    constructor(client:Client) {
+        if (!!AutoCollectClientRequests.INSTANCE) {
+            throw new Error("Client request tracking should be configured from the applicationInsights object");
+        }
+
+        AutoCollectClientRequests.INSTANCE = this;
+        this._client = client;
+    }
+
+    public enable(isEnabled:boolean) {
+        this._isEnabled = isEnabled;
+        if (this._isEnabled && !this._isInitialized) {
+            this._initialize();
+        }
+    }
+
+    public isInitialized() {
+        return this._isInitialized;
+    }
+
+    private _initialize() {
+        this._isInitialized = true;
+        this._initializeHttpModule(http);
+        this._initializeHttpModule(https);
+    }
+
+    private _initializeHttpModule(httpModule: any) {
+        const originalRequest = httpModule.request;
+        httpModule.request = (options, ...requestArgs) => {
+            const request: http.ClientRequest = originalRequest.call(
+                    originalRequest, options, ...requestArgs);
+            if (request) {
+                AutoCollectClientRequests.trackRequest(this._client, options, request);
```

Comment:
Possible, but I think that would make the code less consistent overall.

This dependency request code is following the same pattern as the already existing server request code. That code does not call the [`Client.trackRequest`](https://github.com/Microsoft/ApplicationInsights-node.js/blob/cb64f4a5048a660da0d21c5113e608c8e9083d7c/Library/Client.ts#L138) method either: [the helper constructs a RequestData object directly instead](https://github.com/Microsoft/ApplicationInsights-node.js/blob/cb64f4a5048a660da0d21c5113e608c8e9083d7c/AutoCollection/RequestDataHelper.ts#L69).